core/src/worker/workflow/managed_run.rs

```diff
@@ -1039,13 +1039,26 @@ impl ManagedRun {
     }
 
     fn reply_to_complete(
-        &self,
+        &mut self,
         outcome: ActivationCompleteOutcome,
         chan: Option<oneshot::Sender<ActivationCompleteResult>>,
     ) {
         if let Some(chan) = chan {
-            chan.send(self.build_activation_complete_result(outcome))
-                .expect("Rcv half of activation reply not dropped");
+            if chan
+                .send(self.build_activation_complete_result(outcome))
+                .is_err()
+            {
```

Comment:
I did see this happen once running things against cloud - but I wouldn't be too worried about it. I think it's actually likely happening because of a rare problem in the Rust SDK where it might just drop a future that is calling an activation completion suddenly, thus triggering this problem. AFAICT that's kinda the only way it'd happen.

In any case, it's mitigated now at minimum.